benches/benches/state.rs

```diff
@@ -101,7 +103,68 @@ fn state_single_contract(c: &mut Criterion) {
     group.finish();
 }
 
-fn state_multiple_contracts(c: &mut Criterion) {
+fn insert_state_single_contract_transaction(c: &mut Criterion) {
+    let mut rng = StdRng::seed_from_u64(0xF00DF00D);
+    let state: Bytes32 = rng.gen();
+    let value: Bytes32 = rng.gen();
+
+    let mut bench_state = |group: &mut BenchmarkGroup<WallTime>, name: &str, n: usize| {
+        group.bench_function(name, |b| {
+            let mut db = VmDatabase::default();
+            let contract: ContractId = rng.gen();
+            let mut outer = db.database_mut().transaction();
+            setup(outer.as_mut(), &contract, n);
+            b.iter_custom(|iters| {
+                let mut elapsed_time = Duration::default();
+                for _ in 0..iters {
+                    let mut inner = outer.transaction();
+                    let mut inner_db = VmDatabase::new::<GeneratedConsensusFields>(
```

Comment:
Originally, I was exploring if I could implement `From<DatabaseTransaction>` for `VmDatabase` to simplify this logic. The implementation meant using the default for `GeneratedConsensusFields`, but that is only available under the `test` or `test-helpers` configuration. During benchmarking, neither of this flags are available, and I did not think it would be valuable to extend the list of test configurations to include a benchmarking configuration at this time.